test/test_ion_symbol.cpp

```diff
@@ -22,103 +22,93 @@
+    ION_ASSERT_OK(ion_catalog_open(&catalog));
+    ION_ASSERT_OK(ion_symbol_table_open_with_type(&writer_imports[0], catalog, ist_SHARED));
+    ION_ASSERT_OK(ion_symbol_table_open_with_type(&writer_imports[1], catalog, ist_SHARED));
+    import1 = writer_imports[0];
+    import2 = writer_imports[1];
+    ION_ASSERT_OK(ion_symbol_table_set_name(import1, &import1_name));
+    ION_ASSERT_OK(ion_symbol_table_set_name(import2, &import2_name));
+    ION_ASSERT_OK(ion_symbol_table_set_version(import1, 1));
+    ION_ASSERT_OK(ion_symbol_table_set_version(import2, 1));
+    ION_ASSERT_OK(ion_symbol_table_add_symbol(import1, &sym1, &sid));
+    ION_ASSERT_OK(ion_symbol_table_add_symbol(import2, &sym2, &sid));
+    ION_ASSERT_OK(ion_symbol_table_add_symbol(import2, &sym3, &sid));
+    ION_ASSERT_OK(ion_catalog_add_symbol_table(catalog, import1));
+    ION_ASSERT_OK(ion_catalog_add_symbol_table(catalog, import2));
+    *pcatalog = catalog;
+}
 
-#define ION_SYMBOL_TEST_REWRITE_WITH_CATALOG_FROM_BUFFER_AND_ASSERT_TEXT(data, data_len, expected) \
-    _ION_SYMBOL_TEST_REWRITE_FROM_BUFFER_AND_ASSERT_TEXT(_ION_SYMBOL_TEST_REWRITE_AND_ASSERT_WITH_CATALOG, data, data_len, expected);
+void open_writer_with_imports(bool is_binary, hWRITER *pwriter, ION_STREAM **pstream, ION_WRITER_OPTIONS *pwriter_options, ION_SYMBOL_TABLE **imports, SIZE import_count) {
+    hWRITER writer;
+    ION_STREAM *stream;
+    hCATALOG catalog = NULL;
 
-#define ION_SYMBOL_TEST_REWRITE_FROM_BUFFER_AND_ASSERT_TEXT(data, data_len, expected) \
-    _ION_SYMBOL_TEST_REWRITE_FROM_BUFFER_AND_ASSERT_TEXT(_ION_SYMBOL_TEST_REWRITE_AND_ASSERT, data, data_len, expected);
+    ion_event_initialize_writer_options(pwriter_options);
+    ION_ASSERT_OK(ion_writer_options_initialize_shared_imports(pwriter_options));
+    ION_ASSERT_OK(ion_writer_options_add_shared_imports_symbol_tables(pwriter_options, imports, import_count));
 
-#define _ION_SYMBOL_TEST_REWRITE_FROM_WRITER_AND_ASSERT_TEXT(base_macro, expected) \
-    base_macro( \
-        ION_ASSERT_OK(ion_test_writer_get_bytes(writer, stream, &result, &bytes_flushed)), \
-        free(result), FALSE, expected, strlen(expected) \
-    );
+    pwriter_options->output_as_binary = is_binary;
+    pwriter_options->pcatalog = catalog;
 
-#define ION_SYMBOL_TEST_REWRITE_WITH_CATALOG_FROM_WRITER_AND_ASSERT_TEXT(expected) \
-    _ION_SYMBOL_TEST_REWRITE_FROM_WRITER_AND_ASSERT_TEXT(_ION_SYMBOL_TEST_REWRITE_AND_ASSERT_WITH_CATALOG, expected);
+    ION_ASSERT_OK(ion_stream_open_memory_only(&stream));
+    ION_ASSERT_OK(ion_writer_open(&writer, stream, pwriter_options));
+    *pwriter = writer;
+    *pstream = stream;
+}
 
-#define ION_SYMBOL_TEST_REWRITE_FROM_WRITER_AND_ASSERT_TEXT(expected) \
-    _ION_SYMBOL_TEST_REWRITE_FROM_WRITER_AND_ASSERT_TEXT(_ION_SYMBOL_TEST_REWRITE_AND_ASSERT, expected);
+void rewrite_with_catalog(hCATALOG catalog, bool as_binary, BYTE *data, SIZE data_size, BYTE **p_output, SIZE *p_output_len) {
```

Comment:
PR Tour 🛫:  `rewrite_with_catalog` handles a common pattern in the unit tests where the output of a writer is re-written using the provided catalog. This handles the rewriting part of the `_*REWRITE_*` macros, but does not implement any of the assertions.